client/ayon_substancedesigner/api/project_creation.py

```diff
@@ -147,35 +161,60 @@ def create_project_with_from_template(project_settings=None):
                 template_filepath = get_template_filename_from_project(
                     resources_dir, project_template
                 )
+        elif project_template_setting["template_type"] == (
+            "custom_template"
+            ):
+                custom_template = project_template_setting["custom_template"]
+                project_template = custom_template["custom_template_graph"]
+                if not project_template:
+                    log.warning("Project template not filled. "
+                                "Skipping project creation.")
+                    continue
+
+                path = custom_template["custom_template_path"]
+                if not path:
+                    log.warning("Template path not filled. "
+                                "Skipping project creation.")
+                    continue
+                folder_entity, task_entity = get_entity_data(project_name)
+                template_filepath = resolve_template_path(
+                    path, project_name, folder_entity, task_entity
+                )
+                if not os.path.exists(template_filepath):
+                    log.warning("Template path does not exist yet. "
+                                "Skipping project creation.")
+                    continue
         else:
-            custom_template = project_template_setting["custom_template"]
-            project_template = custom_template["custom_template_graph"]
-            if not project_template:
-                log.warning("Project template not filled. "
+            task_type_template = project_template_setting["task_type_template"]
+            folder_entity, task_entity = get_entity_data(project_name)
+            task_type = task_entity["taskType"]
+            if task_type not in task_type_template["task_types"]:
+                log.warning("Incorrect task types for the project. "
                             "Skipping project creation.")
                 continue
 
-            template_filepath = custom_template["custom_template_path"]
-            if not template_filepath:
-                log.warning("Template path not filled. "
-                            "Skipping project creation.")
-                continue
+            path = task_type_template["path"]
+            template_filepath = resolve_template_path(
+                path, project_name, folder_entity, task_entity)
             if not os.path.exists(template_filepath):
-                log.warning("Template path does not exist yet. "
-                            "Skipping project creation.")
+                log.warning(f"{template_filepath} not found.")
```

Comment:
Be descriptive about what file it is that is not found.
```suggestion
                log.warning(f"Template filepath '{template_filepath}' not found.")
```